src/content/docs/en/reference/configuration-reference.mdx

```diff
@@ -1635,3 +1635,220 @@ The example below displays a generic avatar as fallback content, then animates i
+    const data = await response.json();
+    // Must return an array of entries with an id property,
+    // or an object with IDs as keys and entries as values
+    return data.map((country) => ({
+      id: country.cca3,
+      ...country,
+    }));
+  },
+  // optionally add a schema
+  // schema: z.object...
+});
+
+export const collections = { countries };
+```
+
+For more advanced loading logic, you can define an object loader. This allows incremental updates and conditional loading while also giving full access to the data store. See the API in [the Content Layer API RFC](https://github.com/withastro/roadmap/blob/content-layer/proposals/0047-content-layer.md#loaders).
+
+#### Migrating an existing content collection to use the Content Layer API
+
+You can convert an existing content collection with Markdown, MDX, Markdoc, or JSON entries to use the Content Layer API.
+
+1. **Move the collection folder out of `src/content/`** (e.g. to `src/data/`). All collections located in the `src/content/` folder will use the existing Content Collections API.
+
+    **Do not move the existing `src/content/config.ts` file**. This file will define all collections, using either API.
+
+2. **Edit the collection definition**. Your updated collection requires a `loader`, and the option to select a collection `type` is no longer available.
+
+    ```astro ins={3,8} del={7}
+    // src/content/config.ts
+    import { defineCollection, z } from 'astro:content';
+    + import { glob } from 'astro/loaders';
+
+    const blog = defineCollection({
+      // For content layer you no longer define a `type`
+    - type: 'content',
+    + loader: glob({ pattern: "**\/*.md", base: "./src/data/blog" }),
+      schema: z.object({
+        title: z.string(),
+        description: z.string(),
+        pubDate: z.coerce.date(),
+        updatedDate: z.coerce.date().optional(),
+      }),
+    });
+    ```
+
+3. **Change references from `slug` to `id`**. Content layer collections do not have a `slug` field. Instead, all updated collections will have an `id`.
+
+    ```diff
+    // src/pages/index.astro
+    ---
+    export async function getStaticPaths() {
+      const posts = await getCollection('blog');
+      return posts.map((post) => ({
+    -   params: { slug: post.slug },
+    +   params: { slug: post.id },
+        props: post,
+      }));
+    }
+    ---
+    ```
+
+4. **Switch to the new `render()` function**. Entries no longer have a `render()` method, as they are now serializable plain objects. Instead, import the `render()` function from `astro:content`.
+
+    ```diff
+    // src/pages/index.astro
+    ---
+    - import { getEntry } from 'astro:content';
+    + import { getEntry, render } from 'astro:content';
+
+      const post = await getEntry('blog', params.slug);
+
+    - const { Content, headings } = await post.render();
+    + const { Content, headings } = await render(post);
+    ---
+
+    <Content />
```

Comment:
``````suggestion
    ```astro ins={7} del={6}
    // src/pages/index.astro
    ---
    export async function getStaticPaths() {
      const posts = await getCollection('blog');
      return posts.map((post) => ({
       params: { slug: post.slug },
       params: { slug: post.id },
        props: post,
      }));
    }
    ---
    ```

4. **Switch to the new `render()` function**. Entries no longer have a `render()` method, as they are now serializable plain objects. Instead, import the `render()` function from `astro:content`.

    ```astro ins={4,9} del={3,8}
    // src/pages/index.astro
    ---
    - import { getEntry } from 'astro:content';
    + import { getEntry, render } from 'astro:content';

      const post = await getEntry('blog', params.slug);

    - const { Content, headings } = await post.render();
    + const { Content, headings } = await render(post);
    ---

    <Content />
``````